### ISSUE TITLE
Homebridge-HomeKit link broke after IPv6 tunnel failed?

### ISSUE THREAD

**Issue by Falc0n2k** [homebridge/homebridge#3170]
### Current Situation

Yesterday, for some reason or another, my IPv6 tunnel broke and my entire network reverted back to IPv4. I managed to re-establish the tunnel and get all nodes back up on v6, however, just before bed last night I discovered that HomeKit was unable to connect to my Homebridge installation. There's nothing apparent in the HB logs and all plugins are working fine within HB, which tells me there's an issue with HB and HK connectivity. I checked ufw and the firewall is not running on the Pi (normally this is what causes that issue).

Because of the connectivity problems, I cannot add new HB bridges/devices into HK.

Is there any change any of you have run into this issue in the past or have an idea of where I can start troubleshooting outside of the usual suspects? I even tried MITMproxy to see any connectivity issues and HK isn't issuing any calls outside of my MacBook Air or iPhone. Not sure where to go from there.

### Logs

```markdown
[20/06/2022, 20:23:27] [Nest] Authenticating via access token.
[20/06/2022, 20:23:28] [Nest] Authentication successful.
[20/06/2022, 20:23:30] [Nest] initing thermostat (P) "Entryway Thermostat": deviceId: 641666000096AD4F structureId: fcf62900-ef90-11e8-b70a-12f677e77fbe
[20/06/2022, 20:23:30] [Nest] Temperature unit for Entryway Thermostat is: Fahrenheit
[20/06/2022, 20:23:30] [Nest] Current temperature for Entryway Thermostat is: 25 °C / 77 °F
[20/06/2022, 20:23:30] [Nest] Current humidity for Entryway Thermostat is: 39%
[20/06/2022, 20:23:30] [Nest] Target temperature for Entryway Thermostat is: 25 °C / 77 °F
[20/06/2022, 20:23:30] [Nest] Cooling threshold temperature for Entryway Thermostat is: 24.444 °C / 76 °F
[20/06/2022, 20:23:30] [Nest] Heating threshold temperature for Entryway Thermostat is: 23.333000000000002 °C / 74 °F
[20/06/2022, 20:23:30] [Nest] Online status for Entryway Thermostat is: true
[20/06/2022, 20:23:30] [Nest] Humidity for Entryway Thermostat is: 39
[20/06/2022, 20:23:32] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[20/06/2022, 20:23:36] [HB Supervisor] OS: Linux 5.10.103-v7l+ arm
[20/06/2022, 20:23:36] [HB Supervisor] Node.js v16.15.1 /usr/local/bin/node
[20/06/2022, 20:23:36] [HB Supervisor] Homebridge Path: /usr/local/lib/node_modules/homebridge/bin/homebridge
[20/06/2022, 20:23:36] [HB Supervisor] UI Path: /usr/local/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
[20/06/2022, 20:23:41] [Homebridge UI] Homebridge Config UI X v4.47.0 is listening on :: port 8581
[20/06/2022, 20:23:42] [HB Supervisor] Starting Homebridge with extra flags: -I -D
[20/06/2022, 20:23:42] [HB Supervisor] Started Homebridge v1.4.1 with PID: 1844
[20/06/2022, 20:23:43] Loaded config.json with 0 accessories and 5 platforms.
[20/06/2022, 20:23:43] Loaded 0 cached accessories from cachedAccessories.
[20/06/2022, 20:23:43] ---
[20/06/2022, 20:23:45] Loaded plugin: homebridge-camera-ffmpeg@3.1.4
[20/06/2022, 20:23:45] Registering platform 'homebridge-camera-ffmpeg.Camera-ffmpeg'
[20/06/2022, 20:23:45] ---
[20/06/2022, 20:23:45] Loaded plugin: homebridge-config-ui-x@4.47.0
[20/06/2022, 20:23:45] Registering platform 'homebridge-config-ui-x.config'
[20/06/2022, 20:23:45] ---
[20/06/2022, 20:23:45] Loaded plugin: homebridge-fordpass@1.6.1
[20/06/2022, 20:23:45] Registering platform 'homebridge-fordpass.FordPass'
[20/06/2022, 20:23:45] ---
[20/06/2022, 20:23:45] Loaded plugin: homebridge-nest@4.6.9
[20/06/2022, 20:23:45] Registering platform 'homebridge-nest.Nest'
[20/06/2022, 20:23:45] ---
[20/06/2022, 20:23:45] Loaded plugin: homebridge-shinobi@3.0.9
[20/06/2022, 20:23:45] Registering platform 'homebridge-shinobi.Shinobi'
[20/06/2022, 20:23:45] ---
[20/06/2022, 20:23:46] Disabled plugin: homebridge-simplisafe3@1.10.6
[20/06/2022, 20:23:46] ---
[20/06/2022, 20:23:50] Disabled plugin: homebridge-zigbee-nt@2.5.8
[20/06/2022, 20:23:50] ---
[20/06/2022, 20:23:50] Loading 5 platforms...
[20/06/2022, 20:23:50] [Config] Initializing config platform...
[20/06/2022, 20:23:50] [Config] Running in Service Mode
[20/06/2022, 20:23:50] [Nest] Initializing Nest platform...
[20/06/2022, 20:23:50] [Nest] Initializing child bridge MAC
[20/06/2022, 20:23:50] [Camera FFmpeg] Initializing Camera-ffmpeg platform...
[20/06/2022, 20:23:50] [Camera FFmpeg] Initializing child bridge MAC
[20/06/2022, 20:23:50] Ignoring config for the platform "homebridge-simplisafe3.SimpliSafe 3" in your config.json as the plugin "homebridge-simplisafe3" has been disabled.
[20/06/2022, 20:23:50] Ignoring config for the platform "ZigbeeHomebridgeNTPlatform" in your config.json as the plugin "homebridge-zigbee-nt" has been disabled.
[20/06/2022, 20:23:50] Publishing bridge accessory (name: Homebridge DF2D, publishInfo: {
  username: 'MAC',
  port: 51485,
  pincode: '***-**-***',
  category: 2,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
Setup Payload:
X-HM://0024V4WTYP6R4
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
                       
    ┌────────────┐     
    │ XXX-XX-XXX │     
    └────────────┘     
                       
[20/06/2022, 20:23:50] Homebridge v1.4.1 (HAP v0.10.2) (Homebridge DF2D) is running on port 51485.
[20/06/2022, 20:23:52] [Nest] Launched child bridge with PID 1996
[20/06/2022, 20:23:52] [Camera FFmpeg] Launched child bridge with PID 2002
[20/06/2022, 20:23:52] Registering platform 'homebridge-camera-ffmpeg.Camera-ffmpeg'
[20/06/2022, 20:23:52] [Camera FFmpeg] Loaded homebridge-camera-ffmpeg v3.1.4 child bridge successfully
[20/06/2022, 20:23:52] Loaded 0 cached accessories from cachedAccessories.MAC.
[20/06/2022, 20:23:52] Publishing bridge accessory (name: Camera FFmpeg, publishInfo: {
  username: 'MAC',
  port: 46336,
  pincode: '***-**-***',
  category: 2,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
[20/06/2022, 20:23:52] Registering platform 'homebridge-nest.Nest'
[20/06/2022, 20:23:52] [Nest] Loaded homebridge-nest v4.6.9 child bridge successfully
[20/06/2022, 20:23:52] [Camera FFmpeg] [Front Door] Configuring unbridged accessory...
[20/06/2022, 20:23:52] Loaded 1 cached accessories from cachedAccessories.MAC.
[20/06/2022, 20:23:52] [Camera FFmpeg] [Balcony] Configuring unbridged accessory...
[20/06/2022, 20:23:52] [Camera FFmpeg] [Parking Lot] Configuring unbridged accessory...
[20/06/2022, 20:23:52] Publishing bridge accessory (name: Nest, publishInfo: {
  username: 'MAC',
  port: 53823,
  pincode: '***-**-***',
  category: 2,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
[20/06/2022, 20:23:52] Homebridge v1.4.1 (HAP v0.10.2) (Camera FFmpeg) is running on port 46336.
[20/06/2022, 20:23:52] Publishing external accessory (name: Front Door, publishInfo: {
  username: 'MAC',
  pincode: '***-**-***',
  category: 1,
  port: undefined,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
[20/06/2022, 20:23:52] Publishing external accessory (name: Balcony, publishInfo: {
  username: 'MAC',
  pincode: '***-**-***',
  category: 1,
  port: undefined,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
[20/06/2022, 20:23:52] Publishing external accessory (name: Parking Lot, publishInfo: {
  username: 'MAC',
  pincode: '***-**-***',
  category: 1,
  port: undefined,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'bonjour-hap'
}).
[20/06/2022, 20:23:52] Front Door 8035 is running on port 35229.
[20/06/2022, 20:23:52] Please add [Front Door 8035] manually in Home app. Setup Code: XXX-XX-XXX
[20/06/2022, 20:23:52] Balcony DA6D is running on port 40787.
[20/06/2022, 20:23:52] Please add [Balcony DA6D] manually in Home app. Setup Code: XXX-XX-XXX
[20/06/2022, 20:23:52] [Nest] Fetching Nest devices.
[20/06/2022, 20:23:52] Parking Lot D5CB is running on port 40949.
[20/06/2022, 20:23:52] Please add [Parking Lot D5CB] manually in Home app. Setup Code: XXX-XX-XXX
[20/06/2022, 20:23:52] [Nest] Authenticating via Google.
[20/06/2022, 20:23:52] Homebridge v1.4.1 (HAP v0.10.2) (Nest) is running on port 53823.
[20/06/2022, 20:23:53] [Nest] Authenticating via access token.
[20/06/2022, 20:23:54] [Nest] Authentication successful.
[20/06/2022, 20:23:55] [Nest] initing thermostat (P) "Entryway Thermostat": deviceId: 641666000096AD4F structureId: fcf62900-ef90-11e8-b70a-12f677e77fbe
[20/06/2022, 20:23:55] [Nest] Temperature unit for Entryway Thermostat is: Fahrenheit
[20/06/2022, 20:23:55] [Nest] Current temperature for Entryway Thermostat is: 25 °C / 77 °F
[20/06/2022, 20:23:55] [Nest] Current humidity for Entryway Thermostat is: 39%
[20/06/2022, 20:23:55] [Nest] Target temperature for Entryway Thermostat is: 25 °C / 77 °F
[20/06/2022, 20:23:55] [Nest] Cooling threshold temperature for Entryway Thermostat is: 24.444 °C / 76 °F
[20/06/2022, 20:23:55] [Nest] Heating threshold temperature for Entryway Thermostat is: 23.333000000000002 °C / 74 °F
[20/06/2022, 20:23:55] [Nest] Online status for Entryway Thermostat is: true
[20/06/2022, 20:23:55] [Nest] Humidity for Entryway Thermostat is: 39
```

### Configuration

```json
{
    "bridge": {
        "name": "Homebridge DF2D",
        "username": "MAC",
        "port": 51485,
        "pin": "XXX-XX-XXX",
        "bind": [
            "eth0"
        ],
        "advertiser": "bonjour-hap"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "auth": "form",
            "theme": "dark-mode-blue-grey",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "name": "Nest",
            "refreshToken": "TOKEN",
            "options": [
                "Thermostat.SeparateBuiltInHumiditySensor.Enable",
                "HomeAway.Disable"
            ],
            "_bridge": {
                "username": "MAC",
                "port": 53823
            },
            "platform": "Nest"
        },
        {
            "name": "Camera FFmpeg",
            "cameras": [
                {
                    "name": "Front Door",
                    "manufacturer": "Wyze",
                    "model": "Wyze Cam v2",
                    "serialNumber": "S/N",
                    "firmwareRevision": "4.28.4.49",
                    "unbridge": true,
                    "videoConfig": {
                        "source": "-rtsp_transport tcp -i rtsp://USER:PASS@IP/live",
                        "stillImageSource": "-rtsp_transport tcp -i rtsp://USER:PASS@IP/live -vframes 1 -r 1",
                        "maxStreams": 2,
                        "maxWidth": 1920,
                        "maxHeight": 1080,
                        "maxFPS": 30,
                        "vcodec": "copy",
                        "audio": true
                    }
                },
                {
                    "name": "Balcony",
                    "manufacturer": "Sercomm",
                    "model": "iCamera2",
                    "serialNumber": "S/N",
                    "unbridge": true,
                    "videoConfig": {
                        "source": "-rtsp_transport tcp -i rtsp://USER@IP:554/img/media.sav",
                        "maxWidth": 0,
                        "maxHeight": 0,
                        "audio": true
                    }
                },
                {
                    "name": "Parking Lot",
                    "manufacturer": "Sercomm",
                    "model": "iCamera2",
                    "serialNumber": "S/N",
                    "unbridge": true,
                    "videoConfig": {
                        "source": "-rtsp_transport tcp -i rtsp://USER@IP:554/img/media.sav",
                        "maxWidth": 0,
                        "maxHeight": 0,
                        "audio": true
                    }
                }
            ],
            "platform": "Camera-ffmpeg",
            "_bridge": {
                "username": "MAC",
                "port": 46336
            }
        },
        {
            "name": "Home Alarm",
            "cameras": true,
            "sensorRefresh": 15,
            "persistAccessories": true,
            "cameraOptions": {
                "enableHwaccelRpi": false
            },
            "_bridge": {
                "username": "MAC",
                "port": 40509
            },
            "platform": "homebridge-simplisafe3.SimpliSafe 3"
        },
        {
            "name": "ZigBee",
            "channel": 11,
            "secondaryChannel": "25",
            "adapter": "zstack",
            "routerPollingInterval": 30,
            "disableRouterPolling": false,
            "disableHttpServer": false,
            "disableLed": false,
            "preferAutoDiscover": false,
            "platform": "ZigbeeHomebridgeNTPlatform",
            "_bridge": {
                "username": "MAC",
                "port": 45117
            }
        }
    ],
    "disabledPlugins": [
        "homebridge-simplisafe3",
        "homebridge-zigbee-nt"
    ]
}
```

### Environment

- OS: Raspbian Buster (Linux 5.10.103-v7l+ arm)
- Homebridge: 4.47.0
- Node: v16.15.1
- npm: v8.12.2

### Process Supervisor

hb-service

### Additional Context

_No response
![Screenshot at Jun 20 20-40-20](https://user-images.githubusercontent.com/3246415/174693401-e9785b3c-4e5c-4b7e-adb6-51682a58b878.png)

![Screenshot at Jun 20 20-40-31](https://user-images.githubusercontent.com/3246415/174693413-67e614b2-df89-4293-9d73-97dbe2c8add3.png)

**Comment by Falc0n2k** [comment]
It eventually came back up and was working fine, until the v1.5.0 update was installed today, then everything broke again.

**Comment by Falc0n2k** [comment]
@adriancable -- Thanks for the ideas here. I double checked settings in my router and IGMP traffic is enabled along with mDNS.

> Since unfortunately you don't give us any information about your network set-up, I can't actually tell you what is wrong in your case.

My network is all Ubiquity Unifi gear. My router/firewall is a USG and the downlink is a managed switch (Switch 8) where the Pi is connected directly. Again, there's been no moving of any equipment around to different ports, or any other configuration changes aside from the upgrade to v1.5.0.

**Comment by adriancable** [comment]
@Falc0n2k - how sure are you that there are no "other configuration changes"? Do you have firmware updates disabled everywhere, including on the USG, the managed switch, and the Pi?

Have a look at yourself on Github at the commits/changelog between 1.4.1 and 1.5.0. You will see that nothing in the advertiser code has changed. So, the update to 1.5.0 is not related to your issue. If you need to convince yourself, downgrade back to 1.4.0 to check.

Next is to do some mDNS checks. Get the Discovery app on your iPhone (tildesoft.com), and see if you can see Homebridge under `_hap._tcp` on that. If you can, but HomeKit still can't see the bridge, then HomeKit's internal pairing DB has become corrupted. You'll need to delete the bridge from HomeKit, clear your ~/.homebridge/persist directory and re-add.

If you can't see Homebridge under `_hap._tcp` on your iPhone, then you have an mDNS/IGMP forwarding issue. Many ways to approach this. First confirm that two devices connected to the USG can discover each other, e.g. plug the Pi into the USG directly, plug a Mac into the USG directly, use Discovery on Mac to verify you can see Homebridge advertising on `_hap._tcp` (I'm sure you can). Then progressively move things apart until you can't see it any more. Then you know what's blocking and/or not forwarding mDNS traffic. Then fix that.

No silver bullets or shortcuts here, sorry. You will need to go through the motions to find where the problem occurs. Then you will know what needs fixing.

**Comment by Falc0n2k** [comment]
@adriancable 

> how sure are you that there are no "other configuration changes"? Do you have firmware updates disabled everywhere, including on the USG, the managed switch, and the Pi?

Positive. I don't have firmware updates disabled, however, there hasn't been any updates during the time this incident occurred, so I don't see how that could have impacted anything.

>  So, the update to 1.5.0 is not related to your issue. If you need to convince yourself, downgrade back to 1.4.0 to check.

Did that. No joy. FTR, I never said that v1.5.0 was the definitive cause of the issue, just that it happened at the time things broke, so I drew a logical conclusion based on cause-and-effect.

> Next is to do some mDNS checks. Get the Discovery app on your iPhone (tildesoft.com), and see if you can see Homebridge under _hap._tcp on that. If you can, but HomeKit still can't see the bridge, then HomeKit's internal pairing DB has become corrupted. You'll need to delete the bridge from HomeKit, clear your ~/.homebridge/persist directory and re-add.

Did this also. Again, no joy.

> If you can't see Homebridge under _hap._tcp on your iPhone, then you have an mDNS/IGMP forwarding issue. Many ways to approach this. First confirm that two devices connected to the USG can discover each other, e.g. plug the Pi into the USG directly, plug a Mac into the USG directly, use Discovery on Mac to verify you can see Homebridge advertising on _hap._tcp (I'm sure you can). Then progressively move things apart until you can't see it any more. Then you know what's blocking and/or not forwarding mDNS traffic. Then fix that.

Now this... this I can try and report back.

**Comment by Falc0n2k** [comment]
I managed to lock myself out of the web admin on port 8581, so I decided to enter `hb-config` and see if there was a way to reset the 2FA option. There isn't, but I did see option 3 here:

![image](https://user-images.githubusercontent.com/3246415/176976163-602b96f9-533f-413c-a2e2-24218a3c4b42.png)

After trying this, and `rm auth.json` from `/var/lib/homebridge` and re-creating my user account, I decided almost jokingly to myself to try and add the HB bridge again. Low and behold it worked! I was also able to re-add my other plugins' bridges to HomeKit. I double-checked Discovery and the entry for _hap._tcp is there now as well.
